packages/sales-channel/src/services/sales-channel-module.ts

```diff
@@ -0,0 +1,224 @@
+import {
+  Context,
+  DAL,
+  FilterableSalesChannelProps,
+  FindConfig,
+  InternalModuleDeclaration,
+  ISalesChannelModuleService,
+  ModuleJoinerConfig,
+  RestoreReturn,
+  SalesChannelDTO,
+  SoftDeleteReturn,
+} from "@medusajs/types"
+import {
+  InjectManager,
+  InjectTransactionManager,
+  mapObjectTo,
+  MedusaContext,
+} from "@medusajs/utils"
+import { CreateSalesChannelDTO, UpdateSalesChannelDTO } from "@medusajs/types"
+
+import { SalesChannel } from "@models"
+
+import SalesChannelService from "./sales-channel"
+import {
+  joinerConfig,
+  entityNameToLinkableKeysMap,
+  LinkableKeys,
+} from "../joiner-config"
+
+type InjectedDependencies = {
+  baseRepository: DAL.RepositoryService
+  salesChannelService: SalesChannelService<any>
+}
+
+export default class SalesChannelModuleService<
+  TEntity extends SalesChannel = SalesChannel
+> implements ISalesChannelModuleService
+{
+  protected baseRepository_: DAL.RepositoryService
+  protected readonly salesChannelService_: SalesChannelService<TEntity>
+
+  constructor(
+    { baseRepository, salesChannelService }: InjectedDependencies,
+    protected readonly moduleDeclaration: InternalModuleDeclaration
+  ) {
+    this.baseRepository_ = baseRepository
+    this.salesChannelService_ = salesChannelService
+  }
+
+  __joinerConfig(): ModuleJoinerConfig {
+    return joinerConfig
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async create(
+    data: CreateSalesChannelDTO[],
```

Comment:
sorry, I don't follow. 

I generally support internal services following multiple data approach, and to a certain extent recommend it on entry point services, but wouldn't enforce that on the entry point services.

Reply:
oups my bad, I thought it was the main service 😂

Reply:
oh, but it is right?

Reply:
for the public facing API it can make sense to simplify the API for single actions no?

Reply:
Think we should always support single data and multiple data in the module service i.e. the public-facing service.

E.g.
```ts
async create(data: SomeDTO, sharedContext?: Context: Promise<SomeOtherDTO>
async create(data: SomeDTO[], sharedContext?: Context: Promise<SomeOtherDTO[]>
```

And then, in some cases, a third overload using ID also makes sense e.g. in the cart module, I added:
```ts
async addLineItems(data: SomeDTO, sharedContext?: Context: Promise<CartDTO>
async addLineItems(data: SomeDTO[], sharedContext?: Context: Promise<CartDTO[]>
async addLineItems(cartId: string, data: SomeDTO[], sharedContext?: Context: Promise<CartDTO>
```

There are likely other types of overloads that also make sense, but this will differ from module to module.

As a convention, I think we should always support the first-mentioned example for almost all mutational operations.

Reply:
agreed 👍